ui/library/src/context/DocumentContext.ts

```diff
@@ -5,29 +5,63 @@ import { Dimensions } from '../components/types/boundingBox';
 import { OutlineNode } from '../components/types/outline';
 import { Nullable } from '../components/types/utils';
 import { logProviderWarning } from '../utils/provider';
+import { PageRotation } from '../utils/rotate';
+import { calculateTargetPosition } from '../utils/scroll';
+
+export type OutlinePosition = {
+  pageNumber: number;
+  dest: string;
+  leftPoint: number;
+  bottomPoint: number;
+};
+
+export type OutlinePositionsByPageNumberMap = Map<number, OutlinePosition[]>;
+
+export type OutlineTarget = {
+  dest: string;
+  leftPx: number;
+  topPx: number;
+};
```

Comment:
You can put these types under /types/outline.ts

Reply:
@yensung  done